subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/VerificationFailureHandlingIntegrationTest.groovy

```diff
@@ -0,0 +1,603 @@
+        withCustomTaskInputIndirectlyDependsOnTestTaskOutput()
+
+        expect:
+        fails('customTask')
+        result.assertTaskExecuted(':test')
+        result.assertTaskNotExecuted(':customTask')
+        failure.assertTestsFailed()
+    }
+
+    def 'task does not execute when it has a test task output dependency through a transitive task and test has doFirst that throws --continue'() {
+        given:
+        withTestVerificationFailure()
+        withTestDoFirstActionThrows()
+        withCustomTaskInputIndirectlyDependsOnTestTaskOutput()
+
+        expect:
+        fails('customTask', '--continue')
+        result.assertTaskExecuted(':test')
+        result.assertTaskNotExecuted(':customTask')
+        failure.assertHasCause('intentional failure in Test#doFirst action')
+    }
+
+    def 'task does not execute when it has a test task output dependency through a transitive task and test has doLast that throws --continue'() {
+        given:
+        withPassingTest()
+        withTestDoLastActionThrows()
+        withCustomTaskInputIndirectlyDependsOnTestTaskOutput()
+
+        expect:
+        fails('customTask', '--continue')
+        result.assertTaskExecuted(':test')
+        result.assertTaskNotExecuted(':customTask')
+        failure.assertHasCause('intentional failure in Test#doLast action')
+    }
+
+    def 'task does not execute when it has a test task output dependency through a transitive task and VM exits unexpectedly --continue'() {
+        given:
+        withFatalTestExecutionError()
+        withCustomTaskInputIndirectlyDependsOnTestTaskOutput()
+
+        expect:
+        fails('customTask', '--continue')
+        result.assertTaskExecuted(':test')
+        result.assertTaskNotExecuted(':customTask')
+        assertFatalTestExecutionError()
+    }
+
+    def 'task does not execute when it has a test task output dependency through a transitive task with failing test(s) --continue'() {
+        given:
+        withTestVerificationFailure()
+        withCustomTaskInputIndirectlyDependsOnTestTaskOutput()
+
+        expect:
+        fails('customTask', '--continue')
+        result.assertTaskExecuted(':test')
+        result.assertTaskNotExecuted(':customTask')
+        failure.assertTestsFailed()
+    }
+
+    // helpers
+
+    def withPassingTest() {
+        file('src/test/java/example/PassingUnitTest.java').java '''
+            package example;
+
+            import org.junit.jupiter.api.Test;
+
+            import static org.junit.jupiter.api.Assertions.assertTrue;
+
+            public class PassingUnitTest {
+                @Test
+                public void unitTest() {
+                    assertTrue(true);
+                }
+            }
+        '''
+    }
+
+    /**
+     * Cause the test VM to fail at startup by providing an invalid JVM argument.
+     */
+    def withFatalTestExecutionError() {
+        executer.withStackTraceChecksDisabled() // ignore additional ST only seen on Windows
```

Comment:
Is this still necessary?

Reply:
Yes. The failure to gracefully shut down a test VM adds some unexpected stack traces to the build output, but only for Windows. Should we further parameterize this `withStackTraceChecksDisabled()` call only if the environment is Windows?